sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -509,6 +511,197 @@ public void canAssociateNatGateway() {
+
+        Assertions.assertEquals(2, ipConfigurationInnerList.size());
+    }
+
+    @Test
+    public void canUpdatePrimaryDeleteOption() {
+        String subnetName = generateRandomResourceName("subnet-", 15);
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        Network vnet = networkManager.networks()
+            .define(generateRandomResourceName("vnet-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+            .create();
+
+        nic.update().withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE).apply();
+        nic.refresh();
+        List<NetworkInterfaceIpConfigurationInner> ipConfigurationInnerList =  nic.innerModel().ipConfigurations()
+            .stream().filter(ipConfig -> "primary".equalsIgnoreCase(ipConfig.name()))
+            .collect(Collectors.toList());
+
+        Assertions.assertEquals(1, ipConfigurationInnerList.size());
+        ipConfigurationInnerList.forEach(ipConfigurationInner -> Assertions.assertEquals(DeleteOptions.DELETE, ipConfigurationInner.publicIpAddress().deleteOption()));
+    }
+
+    @Test
+    public void canUpdateNewPrimaryIpAddressWithPrimaryDeleteOption() {
+        String subnetName = generateRandomResourceName("subnet-", 15);
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        Network vnet = networkManager.networks()
+            .define(generateRandomResourceName("vnet-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+            .create();
+
+        nic.update()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
```

Comment:
same as https://github.com/Azure/azure-sdk-for-java/pull/39096/files#r1538628375

Reply:
Merged multiple `update` test cases in the new version.